Added features io

MacU55 opened the pull request:
Changed structure of Main class. Changed methods for reading and writing data from and into file.

MacU55 committed:
changed structure class Main

MacU55 committed:
create readme.md

sdragoff commented on src/com/company/FileUtils.java:
```diff
@@ -2,67 +2,49 @@
-                fileToSave = new File("./results/sortedNumbers.txt");//
-            }
-            fileExist = true;
+    public static List<Integer> unsortedNumbers;
```
remove this

MacU55 replied:
" public static List<Integer> unsortedNumbers;" is removed

sdragoff commented on src/com/company/FileUtils.java:
```diff
@@ -2,67 +2,49 @@
 
-        catch (NumberFormatException t){
+        try {
+            Main.scanner = new Scanner(fileToRead);
```
File utils doesn't know about Main class. scanner= new scanner

MacU55 replied:
Fixed. Class Main  knows classes FileUtils and Sortings. Classes FileUtils and Sortings doesn't knows any classes.

sdragoff commented on src/com/company/FileUtils.java:
```diff
@@ -2,67 +2,49 @@
                 String s = Integer.toString(sortedNumbers.get(i));
                 fileWriter.write(s);
                 fileWriter.write(System.lineSeparator());
+                // fileWriter.close();
```
Remove commented out code

MacU55 replied:
comment is removed

sdragoff commented on src/com/company/Main.java:
```diff
@@ -1,46 +1,82 @@
-import java.util.List;
+import java.nio.file.Files;
 import java.util.Scanner;
+//import static org.junit.Assert.assertTrue;
```
Remove this

MacU55 replied:
comment is removed

sdragoff commented on src/com/company/Main.java:
```diff
@@ -1,46 +1,82 @@
+        scanner = new Scanner(System.in);
+        String fileNameToRead = getFirstParam("Specify path to file with unsorted numbers", scanner);
+        String directoryNameToSaveSortedData = getDirectoryToSave("Specify path to directory to save sorted data", scanner);
+        String fileNameToSaveSortedData = getFileNameToSave("Specify file name to save sorted data", scanner);
```
You have 2 input args, not 3. fileNameToRead and fileNameToSaveSortedData

MacU55 replied:
done

MacU55 replied:
Fixed. There are just two args:  fileNameToRead and fileNameToSaveSortedData

sdragoff commented on src/com/company/Main.java:
```diff
@@ -1,46 +1,82 @@
+        System.out.println(messageUserCanSeeInConsole);
+        return scanner.nextLine();
+    }
+    private static String getFileNameToSave(String messageUserCanSeeInConsole, Scanner scanner){
```
remove this method.

MacU55 replied:
done

MacU55 replied:
method removed

sdragoff commented on src/com/company/Main.java:
```diff
@@ -1,46 +1,82 @@
+                System.out.println("\'" + fileNameToRead + "\' file impossible to read file. Make another readable file with *.txt extension");
+                System.exit(0);
+            }
+            if (!fileToSave.exists()) {
```
you already do this on 45.

MacU55 replied:
fixed

sdragoff commented on src/com/company/Main.java:
```diff
@@ -1,46 +1,82 @@
-
-        if( FileUtils.isFileExists()){
-            FileUtils.getUnsortedNumbers(fileToRead);
+            try(Writer w = new FileWriter("file.log", true)) {
```
Remove this. Save stack to file in another commit

MacU55 replied:
removed

sdragoff commented on src/com/company/Main.java:
```diff
@@ -1,46 +1,82 @@
-            Scanner fileSaveScanner = new Scanner(System.in);
-            String fileSaveSorted = fileSaveScanner.nextLine();
-            fileToSave = new File(fileSaveSorted + ".txt");
+            boolean result = fileToSave.createNewFile();
```
Check if file exists before create.

sdragoff commented on src/com/company/Main.java:
```diff
@@ -1,46 +1,82 @@
 
+            if (!fileToRead.exists()) {
+                System.out.println("\'" + fileNameToRead + "\' file is not found. Check path and file name");
+                System.exit(0);
```
Do not use exit. Read exception throwing.

MacU55 replied:
deleted exit()

sdragoff commented on src/com/company/Sorting.java:
```diff
@@ -3,47 +3,51 @@
-        if (scanSelect.hasNext()){
-            sortMethodNumber = scanSelect.nextInt();
+        try {
+            Main.scanner = new Scanner(System.in);
```
Sorrting doesn't know about Main file.

MacU55 replied:
Fixed. no reference from Sorting to Main

sdragoff commented on src/com/company/Sorting.java:
```diff
@@ -3,47 +3,51 @@
 
+            sortMethodNumber = Main.scanner.nextInt();
+            if (sortMethodNumber == 1) {
+                bubbleSorting(FileUtils.unsortedNumbers);
```
Sorting doesn't know about file utils. Remove it.

MacU55 replied:
Fixed. no reference from Sorting to FileUtils

MacU55 committed:
changed:
1. fixed relationships between classes.
Class Main refer to other classes, but not contraversa.
2. method validateFiles gets two args.
3. added checking condition,  if method validateFiles() is failure - app is closing
